### PR TITLE
[NAE-1690] Fix header responsive overflow mode

### PR DESCRIPTION
**Pull request by machacjozef**
# Description

⚠️ extend constructor with overflowService ⚠️

Fixes [NAE-1690]

## Dependencies

No new dependencies were introduced

### Third party dependencies

No new dependencies were introduced

### Blocking Pull requests

There are no dependencies on other PR

## How Has Been This Tested?

This was tested manually.

### Test Configuration

| Name                | Tested on |
|---------------------| --------- |
| OS                  |      Linux Mint 20.3     |
| Runtime             |  Node v12.22.12         |
| Dependency Manager  |   NPM 6.14.16    |
| Framework version   |    Angular 13.3.1     |
| Run parameters      |           |
| Other configuration |           |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes have been checked, personally or remotely, with @SamuelPalaj 
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes:
    - [x] Lint test
    - [x] Unit tests
    - [x] Integration tests
- [x] I have checked my contribution with code analysis tools:
    - [x] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-engine)
    - [x] [Snyk](https://app.snyk.io/org/netgrif)
- [x] I have made corresponding changes to the documentation:
    - [x] Developer documentation
    - [x] User Guides
    - [x] Migration Guides

[NAE-1690]: https://netgrif.atlassian.net/browse/NAE-1690?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: projects/netgrif-components-core/src/lib/header/abstract-header-service.ts [hunk 1]
```diff
@@ -38,6 +39,7 @@ export abstract class AbstractHeaderService implements OnDestroy {
     protected constructor(protected _headerType: HeaderType,
                           protected _preferences: UserPreferenceService,
                           @Optional() private _viewIdService: ViewIdService,
+                          @Optional() protected _overflowService: OverflowService,
                           protected _logger: LoggerService) {
```
Comment (renczesstefan) [inline projects/netgrif-components-core/src/lib/header/abstract-header-service.ts:43]: Would not be a problem that the last dependency is not optional?

File: projects/netgrif-components-core/src/lib/panel/task-panel/abstract-task-panel.component.ts [hunk 2]
```diff
@@ -342,6 +344,10 @@ export abstract class AbstractTaskPanelComponent extends AbstractPanelWithImmedi
             ? this.taskPanelData.task.finishTitle : 'tasks.view.finish';
     }
 
+    get overflowMode(): boolean {
```
Comment (renczesstefan) [inline projects/netgrif-components-core/src/lib/panel/task-panel/abstract-task-panel.component.ts:347]: `AbstractTaskPanelComponent` and `AbstractCasePanelComponent` are extending `AbstractPanelWithImmediateDataComponent`, that extends `AbstractPanelWithHeaderBindingComponent`. Would not be a good option to extract this method to `AbstractPanelWithHeaderBindingComponent`? Duplicates can be avoided by this.

**Review by renczesstefan — DISMISSED** [review]

File: projects/netgrif-components-core/src/lib/header/task-header/task-header.service.ts [hunk 3]
```diff
@@ -19,8 +20,9 @@ export class TaskHeaderService extends AbstractHeaderService implements OnDestro
                 preferences: UserPreferenceService,
                 logger: LoggerService,
                 @Optional() viewIdService: ViewIdService,
+                @Optional() protected overflowService: OverflowService,
```
Comment (minop) [inline projects/netgrif-components-core/src/lib/header/task-header/task-header.service.ts:23]: why is this `protected` if it is passed to `super` ?

File: projects/netgrif-components-core/src/lib/header/workflow-header/workflow-header.service.ts [hunk 4]
```diff
@@ -6,12 +6,16 @@ import {UserPreferenceService} from '../../user/services/user-preference.service
...
+import {OverflowService} from '../services/overflow.service';
 
 @Injectable()
 export class WorkflowHeaderService extends AbstractHeaderService implements OnDestroy {
 
-    constructor(preferences: UserPreferenceService, @Optional() viewIdService: ViewIdService, logger: LoggerService) {
-        super(HeaderType.WORKFLOW, preferences, viewIdService, logger);
+    constructor(preferences: UserPreferenceService,
+                @Optional() viewIdService: ViewIdService,
+                @Optional() protected overflowService: OverflowService,
```
Comment (minop) [inline projects/netgrif-components-core/src/lib/header/workflow-header/workflow-header.service.ts:16]: `protected` and passed to `super` why?

**Review by minop — CHANGES_REQUESTED** [review]

File: projects/netgrif-components-core/src/lib/header/abstract-header-service.ts (hunk 1 above)
Comment (machacjozef) [reply projects/netgrif-components-core/src/lib/header/abstract-header-service.ts]: Done

File: projects/netgrif-components-core/src/lib/panel/task-panel/abstract-task-panel.component.ts (hunk 2 above)
Comment (machacjozef) [reply projects/netgrif-components-core/src/lib/panel/task-panel/abstract-task-panel.component.ts]: Done

File: projects/netgrif-components-core/src/lib/header/task-header/task-header.service.ts (hunk 3 above)
Comment (machacjozef) [reply projects/netgrif-components-core/src/lib/header/task-header/task-header.service.ts]: Done

File: projects/netgrif-components-core/src/lib/header/workflow-header/workflow-header.service.ts (hunk 4 above)
Comment (machacjozef) [reply projects/netgrif-components-core/src/lib/header/workflow-header/workflow-header.service.ts]: Done

**Review by minop — APPROVED** [review]

File: projects/netgrif-components/src/lib/header/header.theme.scss [hunk 5]
```diff
@@ -14,4 +14,8 @@
     .header-color-font {
         color: mat.get-contrast-color-from-palette($primary, 500);
     }
+
```
Comment (Kovy95) [inline projects/netgrif-components/src/lib/header/header.theme.scss:17]: please write there a wrapper css class

Comment (machacjozef) [reply projects/netgrif-components/src/lib/header/header.theme.scss:17]: Done

**Review by Kovy95 — APPROVED** [review]

**Review by mladoniczky — APPROVED** [review]

**Pull request merged**